datafusion/src/physical_plan/aggregates.rs

```diff
@@ -122,6 +125,7 @@ pub fn return_type(
             coerced_data_types[0].clone(),
             true,
         )))),
+        AggregateFunction::ApproxQuantile => Ok(DataType::Float64),
```

Comment:
```suggestion
        AggregateFunction::ApproxQuantile => Ok(&coerced_data_types[0]),   
```
Make consistent.

Reply:
Tdigest always returns a float, but I can cast this back to the original type - [snowflake](https://docs.snowflake.com/en/sql-reference/functions/approx_percentile.html) always returns a double/float64 for this method, but [Google's BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators), [AWS's RedShift](https://docs.aws.amazon.com/redshift/latest/dg/r_APPROXIMATE_PERCENTILE_DISC.html) and [Postgres](https://www.postgresql.org/docs/9.4/functions-aggregate.html#FUNCTIONS-ORDEREDSET-TABLE) returns the original input type.

I guess most people return the input expr type - happy to do the same - I will push a commit soon 👍

Reply:
Done!

Reply:
> 

I don't find `ApproxQuantile` function in PG data, the link https://www.postgresql.org/docs/9.4/functions-aggregate.html#FUNCTIONS-ORDEREDSET-TABLE provided by you